query-creator/create-query/steps/token-service.js

```diff
@@ -1,3 +1,39 @@
+const fetch = require("node-fetch");
+const TokenServiceClient = require("@concord-consortium/token-service").TokenServiceClient;
+
+const tokenServiceEnv = "staging";
+const resourceType = "athenaWorkgroup";
+const resourceTool = "researcher-report";
+
+exports.findOrCreateResource = async (firebaseJwt, email, portalUrl) => {
+  const client = new TokenServiceClient({ jwt: firebaseJwt, env: tokenServiceEnv, fetch});
+
+  let resource;
+  const escapedEmail = email.replace(/[^\w]/g,'-');
+
+  const allResources = await client.listResources({
```

Comment:
What kind of error-handing were you thinking? This is currently in a nested async/await function. If `listResources` throws an error, it will get re-thrown by `findOrCreateResource` and caught by `app`, which will print out the error trace in the response.

Reply:
That is probably good enough then. I just wanted to make sure users would see the error if one was raised.